corgidrp/darks.py

```diff
@@ -50,61 +51,84 @@ def mean_combine(image_list, bpmap_list, err=False):
-        if bp.dtype != int:
-            raise TypeError('bpmap_list must be made up of int arrays')
-        if bp.shape != s0:
-            raise TypeError('all input list elements must be the same shape')
-        pass
-
-
-    # Get masked arrays
-    ims_m = np.ma.masked_array(image_list, bpmap_list)
+    # uncomment for RAM check
+    # import psutil
+    # process = psutil.Process()
+
+    if not isinstance(dataset_or_image_list, data.Dataset):
+        # if input is an np array or stack, try to accommodate
+        if type(dataset_or_image_list) == np.ndarray:
+            if dataset_or_image_list.ndim == 1: # pathological case of empty array
+                dataset_or_image_list = list(dataset_or_image_list)
+            elif dataset_or_image_list.ndim == 2: #covers case of single 2D frame
+                dataset_or_image_list = [dataset_or_image_list]
+            elif dataset_or_image_list.ndim == 3: #covers case of stack of 2D frames
+                dataset_or_image_list = list(dataset_or_image_list)
+        if type(bpmap_list) == np.ndarray:
+            if bpmap_list.ndim == 1: # pathological case of empty array
+                bpmap_list = list(bpmap_list)
+            elif bpmap_list.ndim == 2: #covers case of single 2D frame
+                bpmap_list = [bpmap_list]
+            elif bpmap_list.ndim == 3: #covers case of stack of 2D frames
+                bpmap_list = list(bpmap_list)
+
+        # Check inputs
+        if not isinstance(bpmap_list, list):
+            raise TypeError('bpmap_list must be a list')
+        if len(dataset_or_image_list) != len(bpmap_list):
+            raise TypeError('image_list and bpmap_list must be the same length')
+        if len(dataset_or_image_list) == 0:
+            raise TypeError('input lists cannot be empty')
+        s0 = dataset_or_image_list[0].shape
+        for index, im in enumerate(dataset_or_image_list):
+            check.twoD_array(im, 'image_list[' + str(index) + ']', TypeError)
+            if im.shape != s0:
+                raise TypeError('all input list elements must be the same shape')
+            pass
+        for index, bp in enumerate(bpmap_list):
+            check.twoD_array(bp, 'bpmap_list[' + str(index) + ']', TypeError)
+            if np.logical_and((bp != 0), (bp != 1)).any():
+                raise TypeError('bpmap_list elements must be 0- or 1-valued')
+            if bp.dtype != int:
+                raise TypeError('bpmap_list must be made up of int arrays')
+            if bp.shape != s0:
+                raise TypeError('all input list elements must be the same shape')
+            pass
 
     # Add non masked elements
-    sum_im = np.zeros_like(image_list[0])
-    map_im = np.zeros_like(image_list[0], dtype=int)
-    for im_m in ims_m:
+    if isinstance(dataset_or_image_list, data.Dataset):
+        with fits.open(dataset_or_image_list[0].filepath, 'readonly') as temp_fits:
```

Comment:
can you just use the Image() class to load in the file? This way, we are reusing code to open/close files. You can access the data and err arrays from the Image attribute.